engine/source/private/core/fileio.cpp

```diff
@@ -0,0 +1,98 @@
+﻿#include "core/fileio.hpp"
+
+#include "core/log.hpp"
+
+#include <filesystem>
+#include <fstream>
+#include <sstream>
+
+namespace lumina
+{
+    FileIO::FileIO()
+    {
+        directories[Directory::Assets] = "engine/assets/";
+        directories[Directory::Config] = "engine/config/";
+        directories[Directory::Log]    = "log/";
+    }
+
+    std::string FileIO::ReadTextFile(Directory directory, const std::string& filePath)
+    {
+        const auto fullPath = GetFilePath(directory, filePath);
+        const std::ifstream file(fullPath);
+        if (!file.is_open())
+        {
+            Log::Error("FileIO::ReadTextFile: Failed to open file {}", fullPath);
+            return "";
+        }
+
+        std::stringstream buffer;
+        buffer << file.rdbuf();
+        return buffer.str();
+    }
+
+    bool FileIO::WriteTextFile(Directory directory, const std::string& filePath, const std::string& data)
+    {
+        const std::string fullPath = GetFilePath(directory, filePath);
+        std::ofstream file(fullPath);
+        if (!file.is_open())
+        {
+            Log::Error("FileIO::WriteTextFile: Failed to open file {}", fullPath);
+            return false;
+        }
+
+        file << data;
```

Comment:
does to overwrite the opened file, or append to the opened file?

Reply:
Overwrites